oci/spec_opts.go

```diff
@@ -1324,6 +1324,19 @@ func WithLinuxDevices(devices []specs.LinuxDevice) SpecOpts {
 	}
 }
 
+func WithLinuxDeviceSymlink(path, permissions string) SpecOpts {
+	resolvedPath, err := filepath.EvalSymlinks(path)
+	if err != nil {
```

Comment:
This is slightly different than the previous iteration, as this code will be called when _setting_ the option, not when it's evaluated. Not sure it's a big difference, but it means that there's potentially time between.

Reply:
I don't think this should be a big difference. More or less just a change in where the evaluation is done. Previous iteration had that in `DeviceFromPath`, this iteration has it in the option itself immediately before calling `DeviceFromPath` on the path.

Reply:
Yes, in most cases it would be fine; there could be use-cases where a construct like this would be used;

```go
var defaultOptions []containerd.SpecOpts{containerd.WithLinuxDeviceFollowSymlinks(.....), containerd.WithFoo()...}
```

In which case the path would be evaluated at the wrong moment.


Either way; thanks for updating 👍